src/util/psbt/mod.rs

```diff
@@ -72,6 +72,24 @@ pub struct PartiallySignedTransaction {
 }
 
 impl PartiallySignedTransaction {
+    /// Returns an iterator for the funding UTXOs of the psbt
+    /// function throws error if transaction inputs have unequal lengths
+    pub fn get_funding_utxos(&self) -> impl Iterator<Item = Result<&TxOut, Error>> {
+        assert_eq!(self.inputs.len(), self.unsigned_tx.input.len());
+        self.unsigned_tx.input.iter().zip(&self.inputs).map(|(tx_input, psbt_input)| {
+            if let Some(ref witness_utxo) = psbt_input.witness_utxo {
```

Comment:
Note: you can avoid `ref` by borrowing: `if let Some(witness_utxo) = &psbt_input.witness_utxo`
I personally prefer this style and I think it's used in this library but not a big deal. `ref` has it's own appeal too.

Also perhaps using match on the whole thing would be clearer:

```rust
match (&psbt_input.witness_utxo, &psbt_input.non_witness_utxo) {
    (Some(witness_utxo), _) => Ok(witness_utxo),
    (None, Some(non_witness_utxo) => {
        ...
    },
    (None, None) => Err(Error::MissingUtxo),
}
```

Reply:
> Thanks, the code looks almost good but we really have to fix two more issues I found.
> 
> There's another thing: should we use a dedicated error type? I tend towards yes but it can be done later as well. I suggest @violet360 if it's too overwhelming to you to add another error type leave it for later, if you feel like adding it you can do it within this PR.
> 
> FTR, this PR will most likely **not** go to the Taproot release anyway.

yes I too was thinking about generating a seperate error type for psbts but thought it would be an overkill, will try committing that in this PR itself.....

Reply:
Note that it still makes sense to also have the variants in `Error` and `From<PsbtUtxoError> for Error`

Reply:
yes,got it, btw, just to be on the same page by a dedicated error type you mean a seperate enum , PsbtUtxoError maybe, in error.rs right ? atleast that's what I think.
just like other error types.

Reply:
It will be cool to have seperate error type for psbt, currently we only have two error types in psbt section....`PsbtUtxoOutOfbounds` and `MissingUtxo` declared in error, and perhaps we can club `pub enum PsbtHash` too inside `psbtUtxoError`

Reply:
should I address it here, or club it up with #837 ?